app/blueprints/assessments/routes.py

```diff
@@ -411,36 +362,493 @@ def get_sorted_application_overviews(application_overviews, column, reverse=Fals
+        flask_app=current_app,
+    )
+
+    # The first call is to get the location data such as country, region and local_authority
+    # from all the existing applications (i.e without search parameters as we don't want to filter
+    # the stats at all).  see https://dluhcdigital.atlassian.net/browse/FS-3249
+    future_all_applications_metadata = thread_executor.submit(
+        get_application_overviews, fund.id, round.id, ""
     )
 
+    # Get all the users for the fund
+    future_users_for_fund = thread_executor.submit(
+        get_users_for_fund,
+        fund.short_name,
+        None,  # round_short_name,
+        True,
+        False,
+    )
+
+    all_applications_metadata = future_all_applications_metadata.result()
+    users_for_fund = future_users_for_fund.result()
+
+    if assessor_role.lower() == "lead_assessor":
+        selectable_users = [
+            user
+            for user in users_for_fund
+            if any("LEAD_ASSESSOR" in role for role in user["roles"])
+        ]
+    else:
+        selectable_users = [
+            user
+            for user in users_for_fund
+            if any(
+                "ASSESSOR" in role and "LEAD_ASSESSOR" not in role
+                for role in user["roles"]
+            )
+        ]
+
+    unfiltered_stats = process_assessments_stats(all_applications_metadata)
+
+    return render_template(
+        "select_assessor.html",
+        selected_assessments=selected_assessments,
+        assessor_role=assessor_role,
+        round_details=round_details,
+        stats=unfiltered_stats,
+        users=selectable_users,
+        selected_users=selected_users,
+        form=form,
+    )
+
+
+@assessment_bp.route(
+    "/assign/<fund_short_name>/<round_short_name>/overview",
+    methods=["POST"],
+)
+@check_access_fund_short_name_round_sn
+def assignment_overview(fund_short_name: str, round_short_name: str):
+    if "change_users" in request.form:
+        return redirect(
+            url_for(
+                "assessment_bp.assessor_type_list",
+                fund_short_name=fund_short_name,
+                round_short_name=round_short_name,
+            ),
+            307,
+        )
+
+    if "change_roles" in request.form:
+        return redirect(
+            url_for(
+                "assessment_bp.assessor_type",
+                fund_short_name=fund_short_name,
+                round_short_name=round_short_name,
+            ),
+            307,
+        )
+
+    if "change_assessments" in request.form:
+        return redirect(
+            url_for(
+                "assessment_bp.assign_assessments",
+                fund_short_name=fund_short_name,
+                round_short_name=round_short_name,
+            ),
+            307,
+        )
+
+    selected_assessments = request.form.getlist("selected_assessments")
+    assessor_role = request.form.getlist("assessor_role")[0]
+    selected_user_ids = request.form.getlist("selected_users")
```

Comment:
if the list is empty we might need to handle 
```request.form.getlist("assessor_role")```

Reply:
As above.